hibernate-community-dialects/src/main/java/org/hibernate/community/dialect/H2LegacyDialect.java

```diff
@@ -0,0 +1,707 @@
+		functionFactory.yearMonthDay();
+		functionFactory.hourMinuteSecond();
+		functionFactory.dayOfWeekMonthYear();
+		functionFactory.weekQuarter();
+		functionFactory.daynameMonthname();
+		if ( useLocalTime ) {
+			functionFactory.localtimeLocaltimestamp();
+		}
+		functionFactory.bitLength();
+		functionFactory.octetLength();
+		functionFactory.ascii();
+		functionFactory.octetLength();
+		functionFactory.space();
+		functionFactory.repeat();
+		functionFactory.chr_char();
+		functionFactory.instr();
+		functionFactory.substr();
+		//also natively supports ANSI-style substring()
+		functionFactory.position();
+		functionFactory.trim1();
+		functionFactory.concat_pipeOperator();
+		functionFactory.nowCurdateCurtime();
+		functionFactory.sysdate();
+		functionFactory.insert();
+//		functionFactory.everyAny(); //this would work too
+		functionFactory.everyAny_boolAndOr();
+		functionFactory.median();
+		functionFactory.stddevPopSamp();
+		functionFactory.varPopSamp();
+		if ( getVersion().isSame( 1, 4, 200 ) ) {
+			// See https://github.com/h2database/h2database/issues/2518
+			functionFactory.format_toChar();
+		}
+		else {
+			functionFactory.format_formatdatetime();
+		}
+		functionFactory.rownum();
+		if ( getVersion().isSameOrAfter( 1, 4, 200 ) ) {
+			functionFactory.windowFunctions();
+			functionFactory.listagg( null );
+			if ( getVersion().isSameOrAfter( 2 ) ) {
+				functionFactory.inverseDistributionOrderedSetAggregates();
+				functionFactory.hypotheticalOrderedSetAggregates();
+			}
+		}
+		else {
+			functionFactory.listagg_groupConcat();
+		}
+	}
+
+	@Override
+	public void augmentPhysicalTableTypes(List<String> tableTypesList) {
+		if ( getVersion().isSameOrAfter( 2 ) ) {
+			tableTypesList.add( "BASE TABLE" );
+		}
+	}
+
+	@Override
+	protected Integer resolveSqlTypeCode(String columnTypeName, TypeConfiguration typeConfiguration) {
+		switch ( columnTypeName ) {
+			case "FLOAT(24)":
+				// Use REAL instead of FLOAT to get Float as recommended Java type
+				return Types.REAL;
+		}
+		return super.resolveSqlTypeCode( columnTypeName, typeConfiguration );
+	}
+
+	@Override
+	public JdbcType resolveSqlTypeDescriptor(
+			String columnTypeName,
+			int jdbcTypeCode,
+			int precision,
+			int scale,
+			JdbcTypeRegistry jdbcTypeRegistry) {
+		// As of H2 2.0 we get a FLOAT type code even though it is a DOUBLE
+		if ( jdbcTypeCode == FLOAT && "DOUBLE PRECISION".equals( columnTypeName ) ) {
+			return jdbcTypeRegistry.getDescriptor( DOUBLE );
+		}
+		return super.resolveSqlTypeDescriptor( columnTypeName, jdbcTypeCode, precision, scale, jdbcTypeRegistry );
+	}
+
+	@Override
+	protected Integer resolveSqlTypeCode(String typeName, String baseTypeName, TypeConfiguration typeConfiguration) {
+		switch ( baseTypeName ) {
+			case "CHARACTER VARYING":
+				return VARCHAR;
+		}
+		return super.resolveSqlTypeCode( typeName, baseTypeName, typeConfiguration );
+	}
+
+	@Override
+	public int getMaxVarcharLength() {
+		return 1_048_576;
```

Comment:
The actual limit for non-LOB data type declarations is
2,147,483,647 characters in H2 1.4.200 and older versions
1,048,576 characters in H2 2.0.202–2.1.210.
1,000,000,000 characters in H2 2.1.212 and newer versions.

LOBs allow longer values.

Reply:
PRs are welcome ;)